backend/mr-eventing/src/main/kotlin/io/featurehub/mr/events/converter/FeatureMessagingConverterImpl.kt

```diff
@@ -0,0 +1,106 @@
+        if (defaultValueUpdate?.hasChanged == true && defaultValueUpdated != null && defaultValuePrevious != null) it.featureValueUpdated(
+          MessagingFeatureValueUpdate()
+            .valueType(featureValue.feature.valueType)
+            .updated(defaultValueUpdated)
+            .previous(defaultValuePrevious)
+        )
+        else it
+      }
+      .let {
+        val lockUpdated = featureMessagingParameter.lockUpdate?.updated
+        val lockPrevious = featureMessagingParameter.lockUpdate?.previous
+        if (featureMessagingParameter.lockUpdate?.hasChanged == true && lockUpdated != null && lockPrevious != null)
+          it.lockUpdated(MessagingLockUpdate()
+            .updated(lockUpdated)
+            .previous(lockPrevious)
+        ) else it
+      }
+      .let {
+        val retiredUpdated = featureMessagingParameter.retiredUpdate?.updated
+        val retiredPrevious = featureMessagingParameter.retiredUpdate?.previous
+        if (featureMessagingParameter.retiredUpdate?.hasChanged == true && retiredUpdated != null && retiredPrevious != null)
+          it.retiredUpdated(MessagingRetiredUpdate()
+            .updated(retiredUpdated)
+            .previous(retiredPrevious)
+        ) else it
+      }
+      .let {
+        val messagingStrategiesReorder = MessagingStrategiesReorder()
+        val strategyUpdates = featureMessagingParameter.strategyUpdates
+        if (strategyUpdates?.hasChanged == true && strategyUpdates.updated.isNotEmpty())
+          it.strategiesUpdated(strategyUpdates.updated.map { rolloutStrategyUpdate ->  toMessagingStrategyUpdate(rolloutStrategyUpdate) })
+        if (strategyUpdates?.hasChanged == true && strategyUpdates.reordered.isNotEmpty())
+          it.strategiesReordered(
+            messagingStrategiesReorder.reordered(
+          strategyUpdates.reordered.map { rolloutStrategy -> toMessagingRolloutStrategy(rolloutStrategy) }
+        ))
+        if (strategyUpdates?.hasChanged == true && strategyUpdates.previous.isNotEmpty())
+          it.strategiesReordered(
+            messagingStrategiesReorder.previous(
+            strategyUpdates.previous.map { rolloutStrategy -> toMessagingRolloutStrategy(rolloutStrategy) }
+          ))
+        else it
+      }
+
+  }
+
+  private fun toMessagingStrategyUpdate(rolloutStrategyUpdate: RolloutStrategyUpdate): MessagingStrategyUpdate {
+    val new = rolloutStrategyUpdate.new
+    val old = rolloutStrategyUpdate.old
+    return MessagingStrategyUpdate()
+      .let { if (new != null) it.newStrategy(toMessagingRolloutStrategy(new)) else it }
+      .let { if (old != null) it.oldStrategy(toMessagingRolloutStrategy(old)) else it }
+      .updateType(StrategyUpdateType.fromValue(rolloutStrategyUpdate.type))
+  }
+
+  private fun toMessagingRolloutStrategy(rolloutStrategy: RolloutStrategy): MessagingRolloutStrategy {
+    val attributes = rolloutStrategy.attributes
+    return MessagingRolloutStrategy()
+      .id(rolloutStrategy.id ?: "rs-id")
```

Comment:
this should _never_ be true any longer, rollout strategy ids have to have a value otherwise then can't be checked for ordering. If the front end doesn't allocate one, i'm pretty sure the backend forces it to have one? There was a front end bug we fixed last week where it kept changing this :-)

Reply:
Since this field is nullable in the database so I'll have to do a null check here.